src/lib/Form/Type/Trash/TrashItemCheckboxType.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Type\Trash;
+
+use eZ\Publish\API\Repository\TrashService;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use EzSystems\EzPlatformAdminUi\Form\DataTransformer\TrashItemTransformer;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class TrashItemCheckboxType extends AbstractType
+{
+    /**
+     * @var TrashService
+     */
+    private $trashService;
+
+    public function __construct(TrashService $trashService)
+    {
+        $this->trashService = $trashService;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addModelTransformer(new TrashItemTransformer($this->trashService));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars = array_replace($view->vars, [
+            'value' => $form->getViewData(),
+            'checked' => false,
+        ]);
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'compound' => false,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'trash_item_checkbox';
```

Comment:
Is it required? We keep it out of our `Type` classes so that the names are generated from class names.